Userland/Services/SystemServer/Service.cpp

```diff
@@ -144,14 +157,38 @@ void Service::handle_socket_connection()
 
 void Service::activate()
 {
-    VERIFY(m_pid < 0);
+    // Already running
+    if (m_pid >= 0)
+        return;
+    
+    // It makes that the service won't restart automatically.
+    m_force_stopped = false;
```

Comment:
This causes the service to be restarted automatically, not the other way around, doesn't it?